funnel/views/project.py

```diff
@@ -281,6 +298,45 @@ def rsvp(self):
     def rsvp_list(self):
         return {'project': self.obj, 'statuses': RSVP_STATUS}
 
+    @route('save', methods=['POST'])
+    @render_with(json=True)
+    @lastuser.requires_login
+    @requires_permission('view')
+    def save(self):
+        form = SavedProjectForm()
+        if form.validate_on_submit():
+            proj_save = SavedProject.query.filter_by(user=current_auth.user, project=self.obj).first()
+            if form.save.data:
+                if proj_save is None:
+                    proj_save = SavedProject(user=current_auth.user, project=self.obj)
+                    form.populate_obj(proj_save)
+                    db.session.commit()
+                    return {'status': 'ok'}
+                else:
+                    return {
+                        'status': 'error',
+                        'error': 'project_save_exists',
+                        'error_description': _("You have already saved this project")
+                    }
```

Comment:
Remove this. Breaks idempotency. The operation is successful even if no action was required. You could distinguish by returning HTTP status code 201 vs 200, with the same `{'status': 'ok'}` for both.